core/services/relay/evm/functions/contract_transmitter.go

```diff
@@ -25,33 +26,43 @@ import (
 	evmRelayTypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
 )
 
+type roundRobinKeystore interface {
+	GetRoundRobinAddress(ctx context.Context, chainID *big.Int, addresses ...common.Address) (address common.Address, err error)
+}
+
+type txManager interface {
+	CreateTransaction(ctx context.Context, txRequest txmgr.TxRequest) (tx txmgr.Tx, err error)
+}
+
 type FunctionsContractTransmitter interface {
 	services.ServiceCtx
 	ocrtypes.ContractTransmitter
 }
 
-type Transmitter interface {
-	CreateEthTransaction(ctx context.Context, toAddress common.Address, payload []byte, txMeta *txmgr.TxMeta) error
-	FromAddress() common.Address
-}
-
 type ReportToEthMetadata func([]byte) (*txmgr.TxMeta, error)
 
 func reportToEvmTxMetaNoop([]byte) (*txmgr.TxMeta, error) {
 	return nil, nil
 }
 
 type contractTransmitter struct {
-	contractAddress     atomic.Pointer[common.Address]
-	contractABI         abi.ABI
-	transmitter         Transmitter
-	transmittedEventSig common.Hash
-	contractReader      contractReader
-	lp                  logpoller.LogPoller
-	lggr                logger.Logger
-	reportToEvmTxMeta   ReportToEthMetadata
-	contractVersion     uint32
-	reportCodec         encoding.ReportCodec
+	contractAddress             atomic.Pointer[common.Address]
+	contractABI                 abi.ABI
+	transmittedEventSig         common.Hash
+	contractReader              contractReader
+	lp                          logpoller.LogPoller
+	lggr                        logger.Logger
+	reportToEvmTxMeta           ReportToEthMetadata
```

Comment:
We can get rid of this, it's always set to nil (= no-op). Also remove reportToEvmTxMetaNoop function.